ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/execution/JsonRpcExecutorVerticleTest.java

```diff
@@ -0,0 +1,218 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.ethereum.api.handlers.NonBlockingJsonRpcExecutorHandler;
+import org.hyperledger.besu.ethereum.api.jsonrpc.JsonRpcConfiguration;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.opentelemetry.api.trace.Tracer;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.ReplyException;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(VertxExtension.class)
+@ExtendWith(MockitoExtension.class)
+class JsonRpcExecutorVerticleTest {
+  NonBlockingJsonRpcExecutorHandler nonBlockingJsonRpcExecutorHandler;
+
+  @Mock JsonRpcExecutor mockJsonRpcExecutor;
+  @Mock JsonRpcConfiguration mockJsonRpcConfiguration;
+
+  @BeforeEach
+  void setUp(final Vertx vertx) {
+    final JsonRpcExecutorVerticle jsonRpcExecutorVerticle =
+        new JsonRpcExecutorVerticle(
+            mockJsonRpcExecutor, mock(Tracer.class), mockJsonRpcConfiguration);
+
+    nonBlockingJsonRpcExecutorHandler =
+        new NonBlockingJsonRpcExecutorHandler(vertx, List.of(jsonRpcExecutorVerticle));
+    await().atMost(5, TimeUnit.SECONDS).until(() -> vertx.deploymentIDs().size() == 1);
+  }
+
+  @AfterEach
+  void tearDown(final Vertx vertx) {
+    nonBlockingJsonRpcExecutorHandler.stop();
+    await().atMost(5, TimeUnit.SECONDS).until(() -> vertx.deploymentIDs().size() == 0);
+  }
+
+  @Test
+  void correctJsonResponseOfObjectRequestShouldBeReturned(
+      final Vertx vertx, final VertxTestContext testContext) throws Throwable {
+    final JsonRpcResponse expectedResponse = new JsonRpcSuccessResponse("42", "successResult");
+    when(mockJsonRpcExecutor.execute(any(), any(), any(), any(), any(), any()))
+        .thenReturn(expectedResponse);
+    vertx
+        .eventBus()
+        .request(RPC_EXECUTE_OBJECT.getAddress(), mock(JsonRpcExecutorObjectRequest.class))
+        .onSuccess(
+            actualResponse -> {
+              assertThat(actualResponse.body()).isEqualTo(expectedResponse);
+              testContext.completeNow();
+            });
+
+    assertThat(testContext.awaitCompletion(5, TimeUnit.SECONDS)).isTrue();
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+  }
+
+  @Test
+  void exceedingMaxBatchSizeShouldCauseFailure(
```

Comment:
What dictates maxBatch in a production scenario? 
IOW what must happen in a real-world scenario in order to trigger this error?